lite2/store/db/db.go

```diff
@@ -207,14 +207,14 @@ func (s *dbs) FirstSignedHeaderHeight() (int64, error) {
 // height. It returns ErrSignedHeaderNotFound if no such header exists.
 //
 // Safe for concurrent use by multiple goroutines.
-func (s *dbs) SignedHeaderAfter(height int64) (*types.SignedHeader, error) {
+func (s *dbs) SignedHeaderBefore(height int64) (*types.SignedHeader, error) {
 	if height <= 0 {
 		panic("negative or zero height")
 	}
 
-	itr, err := s.db.Iterator(
-		s.shKey(height+1),
-		append(s.shKey(1<<63-1), byte(0x00)),
+	itr, err := s.db.ReverseIterator(
+		nil,
```

Comment:
don't we need to have a stop key?

Reply:
Can you explain what a stop key is and why I need it, I thought using nil was sufficient when I read the comments regarding the reverse iterator

Reply:
suppose you have keys

Y:100
H:100
H:200

...
H300

in your DB

in the absence of H:100, it will scan Y:100. It's ok since we won't include it. It's just extra work we don't really need to do IF WE SET END KEY

Reply:
Am I supposed to replace nil with append(s.shKey(1<<63-1), byte(0x00))

Reply:
s.shKey(1)

Reply:
okay thanks - sorry I'm just not so familiar with the db